build.rs

```diff
@@ -429,7 +429,7 @@ impl Target {
 
 fn build_c_code(target: &Target, out_dir: &Path) {
     let includes_modified = RING_INCLUDES.par_iter()
-        .weight_max()
+        .with_max_len(1)
```

Comment:
`weight_max()` was the same as `weight(INFINITY)`, which effectively meant to split the jobs into the smallest possible size -- a single item.

[`with_max_len(n)`](https://docs.rs/rayon/0.7.0/rayon/iter/trait.IndexedParallelIterator.html#method.with_max_len) basically means to process no more than `n` items in each job.  So `with_max_len(1)` splits it completely, same as `weight_max` did.

In general, we hope that `with_min_len`/`with_max_len` are more intuitive than the rather arbitrary floating point "weight" thresholds were.  `INFINITY` is easy, but for something like `1000.0` it was hard to figure out what rayon would do.